### PR TITLE
raise error for tkr controller if image path is empty

### PR DESCRIPTION
**Pull request by hongtaiwang**
to avoid stack track error when calling `ListImageTags`

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

File: tkr/controller/tkr-source/fetcher/fetcher.go [hunk 1]
```diff
@@ -176,6 +176,9 @@ func (f *Fetcher) saveTKRCompatibilityCM(ctx context.Context, ns string, metadat
 }
 
 func (f *Fetcher) fetchCompatibilityMetadata() (*tkrv1.CompatibilityMetadata, error) {
+	if f.Config.BOMMetadataImagePath == "" {
+		return nil, errors.New("compatibility metadata image path should not be empty")
+	}
```
Comment (imikushin) [inline tkr/controller/tkr-source/fetcher/fetcher.go:181]: Can we insert this check _before_ starting the controller? It feels like, at this point in the code, it's already too late.

So, maybe in main.go where we just got these config values and are creating Config objects.

File: tkr/controller/tkr-source/fetcher/fetcher.go [hunk 2]
```diff
@@ -231,6 +234,10 @@ func (f *Fetcher) fetchTKRBOMConfigMaps(ctx context.Context) error {
 		return err
 	}
 
+	if f.Config.BOMImagePath == "" {
+		return errors.New("bom image should not be empty")
+	}
+
```
Comment (imikushin) [inline tkr/controller/tkr-source/fetcher/fetcher.go:240]: Same.

File: tkr/controller/tkr-source/fetcher/fetcher.go [hunk 3]
```diff
@@ -334,6 +341,10 @@ func (f *Fetcher) fetchTKRPackages(ctx context.Context) error {
 		return err
 	}
 
+	if f.Config.TKRRepoImagePath == "" {
+		return errors.Wrap(err, "failed to list current available TKR Package Repository image tags, should not be empty")
+	}
+
```
Comment (imikushin) [inline tkr/controller/tkr-source/fetcher/fetcher.go:347]: Same.

File: tkr/controller/tkr-source/fetcher/fetcher.go (hunk 1 above)
Comment (hongtaiwang) [reply tkr/controller/tkr-source/fetcher/fetcher.go]: sure. updated

**Review by imikushin — APPROVED** [review]
LGTM. Thanks @hongtaiwang !

**Pull request merged**